plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/explorer/OpenShiftExplorerContentProvider.java

```diff
@@ -44,197 +41,157 @@
  * 
  * @author jeff.cantrill
  */
-public class OpenShiftExplorerContentProvider extends BaseExplorerContentProvider 
-	implements PropertyChangeListener, IProjectCache.IProjectCacheListener {
-	
+public class OpenShiftExplorerContentProvider implements ITreeContentProvider {
 	private static final List<String> TERMINATED_STATUS = Arrays.asList("Complete", "Failed", "Error", "Cancelled");
-	private static final IProjectCache cache = new OpenShiftProjectCache();
-	
+
+	private OpenshiftUIModel model;
+	private IElementListener listener;
+	private StructuredViewer viewer;
+	private Map<Object, BaseExplorerContentProvider.LoadingStub> stubs= new HashMap<Object, BaseExplorerContentProvider.LoadingStub>();
+
 	public OpenShiftExplorerContentProvider() {
-		super();
-		cache.addListener(this);
+		this(new OpenshiftUIModel());
```

Comment:
Not very much yet, but there are calls Connection.getResources() in wizards, handlers, utilities and some of them might be replaced with queries to the model to avoid extra requests for resources to the client. 
For testing reasons, constructor may take not model but model factory, and tests will create content provider with a specific factory.